### ISSUE TITLE
app/store.js cannot extend Store with new methods in beta.19

### ISSUE THREAD

**Issue by cdoornink** [emberjs/data#3239]
This worked in beta.18 and is broken in beta.19:

app/store.js

```
import DS from 'ember-data';

export default DS.Store.extend({
  newMethodName: function() {
    //calling this function gives me: 'error: this.store.newMethodName is not a function'
    return
  }
})
```

Changing from Store.extend to Store.reopen fixes this as well as moving the file to app/application/store.js but @bmac pointed out in Slack that he may have done something to break this so I'm reporting this as a bug here.

**Comment by fivetanley** [comment]
I think this has something to do with the `store:main` => `application:main` change bmac made. Is there a  way we could warn about this in ember cli?

**Comment by bmac** [comment]
@cdoornink I just released `Ember Data beta-19.2` with a fix for this regression.
